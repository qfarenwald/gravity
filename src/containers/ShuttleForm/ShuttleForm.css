.ShuttleForm {
  align-items: center;
  color: white;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.ShuttleForm-input {
  border: none;
  border-radius: 15px;
  font-size: 1rem;
  letter-spacing: 5px;
  padding: 10px;
  margin: 10px;
  width: 50%;
  text-align: center;
}

.ShuttleForm-btn {
  background-color: #38B449;
  border: none;
  border-radius: 25px;
  font-size: 1rem;
  letter-spacing: 5px;
  padding: 20px;
  margin-top: 50px;
  width: 300px;
}

.ShuttleForm-btn:hover {
  background-color: #33993D;
  color: white;
  cursor: pointer;
}

.ShuttleForm-btn:disabled {
  background-color: grey;
  color: black;
  cursor: not-allowed;
}

.error-p {
  color: red;
  font-size: .75rem;
  letter-spacing: 5px;
}

.error-p-hidden {
  color: black;
  font-size: .75rem;
  letter-spacing: 5px;
}
