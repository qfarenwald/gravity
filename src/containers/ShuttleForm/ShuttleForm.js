import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { makeBooking } from '../../actions';
import { Link } from 'react-router-dom';
import './ShuttleForm.css';

export class ShuttleForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: ''
    }
  }

  clearInputs = () => {
    this.setState({
      name: '',
      email: '',
      formReady: false
    })
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
    if(this.state.name !== '' && this.state.email !== '') {
      this.setState({
        formReady: true
      })
    }
  }

  handleClick = (e) => {
    e.preventDefault()
    const { makeBooking } = this.props
    if(this.state.name !== '' && this.state.email !== '') {
      makeBooking(this.state)
      this.clearInputs()
    }
  }

  render() {
    return (
      <form className="ShuttleForm">
        <h3>BOOK YOUR SHUTTLE</h3>
        <input className="ShuttleForm-input"
          name="name"
          type="text"
          placeholder="Enter Name"
          value={this.state.name}
          onChange={this.handleChange}
        />
        <input className="ShuttleForm-input"
          name="email"
          type="text"
          placeholder="Enter Email"
          value={this.state.email}
          onChange={this.handleChange}
        />
        <Link to={this.state.formReady ? "/confirmation" : "/shuttle"}><button type="button" className={this.state.formReady ? "ShuttleForm-btn" : "ShuttleForm-btn-disabled"} onClick={this.handleClick}>SUBMIT BOOKING</button></Link>
        <Link to='/trails' className="link"><h5>BACK TO TRAILS</h5></Link>
      </form>
    )
  }
}

export const mapDispatchToProps = dispatch => (
  bindActionCreators(
    {
      makeBooking
    },
  dispatch)
)

export default connect(null, mapDispatchToProps)(ShuttleForm)
