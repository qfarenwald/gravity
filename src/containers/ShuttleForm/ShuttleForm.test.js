import React from 'react';
import ReactDOM from 'react-dom';
import { ShuttleForm } from './ShuttleForm';
import { shallow } from 'enzyme';

describe('ShuttleForm', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<ShuttleForm />)

    expect(wrapper).toMatchSnapshot()
  })
})
