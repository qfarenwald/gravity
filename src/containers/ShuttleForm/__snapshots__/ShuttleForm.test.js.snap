// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShuttleForm should match snapshot 1`] = `
<form
  className="ShuttleForm"
>
  <h3>
    BOOK YOUR SHUTTLE
  </h3>
  <input
    className="ShuttleForm-input"
    name="name"
    onChange={[Function]}
    placeholder="Enter Name"
    type="text"
    value=""
  />
  <input
    className="ShuttleForm-input"
    name="email"
    onChange={[Function]}
    placeholder="Enter Email"
    type="text"
    value=""
  />
  <input
    className="ShuttleForm-input"
    name="riders"
    onChange={[Function]}
    placeholder="Enter Number Of Riders"
    type="number"
    value=""
  />
  <input
    className="ShuttleForm-input"
    name="bikes"
    onChange={[Function]}
    placeholder="Enter Number Of Bikes"
    type="number"
    value=""
  />
  <input
    className="ShuttleForm-input"
    name="date"
    onChange={[Function]}
    placeholder="Enter Date (mm/dd)"
    type="number"
    value=""
  />
  <input
    className="ShuttleForm-input"
    name="starttime"
    onChange={[Function]}
    placeholder="Enter Start Time (00:00 am/pm)"
    type="number"
    value=""
  />
  <input
    className="ShuttleForm-input"
    name="endtime"
    onChange={[Function]}
    placeholder="Enter End Time (00:00 am/pm)"
    type="number"
    value=""
  />
  <p
    className="ShuttleForm-p"
  >
    PLEASE FILL OUT ALL INPUTS
  </p>
  <button
    className="ShuttleForm-btn"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    SUBMIT BOOKING
  </button>
  <Link
    className="link"
    to="/trails"
  >
    <h5>
      BACK TO TRAILS
    </h5>
  </Link>
</form>
`;
